pkg/prometheus/operator_test.go

```diff
@@ -197,11 +196,11 @@ func TestValidateRemoteWriteConfig(t *testing.T) {
 		test := c
 		t.Run(test.name, func(t *testing.T) {
 			err := ValidateRemoteWriteSpec(test.spec)
-			if err != nil && !test.expectErr {
-				t.Fatalf("unexpected error occurred: %v", err)
+			if !test.expectErr {
+				require.NoError(t, err)
```

Comment:
This might be better. Wdyt?

```
require.False(t,(err!=nil && !test.expectErr),"unexpected error occurred: %v", err)
```

Reply:
I think it looks good the way it is now, we just don't need the second if because, if we get there, it's always true.

What do you think about the suggestion below?

```go

	for _, c := range cases {
		test := c
		t.Run(test.name, func(t *testing.T) {
			err := ValidateRemoteWriteSpec(test.spec)
			if test.expectErr {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
		})
	}
```